### PR TITLE
Minimal accounting state PoC

### PR DESCRIPTION
**Pull request by oskarth**
Addresses https://github.com/status-im/nim-waku/issues/269 to keep accounting state.  Starts to introduce WakuSwap protocol.

Future enhancements:
- End to end https://github.com/status-im/nim-waku/issues/270
- Config flag fix https://github.com/status-im/nim-waku/issues/269
- Better tests (probably as part of 270)

Tested with chat2 as PoC and flag on:

```
INF 2020-11-18 14:47:49.393+08:00 mounting store                             topics="wakunode" tid=10675
INF 2020-11-18 14:47:49.394+08:00 mounting swap                              topics="wakunode" tid=10675
INF 2020-11-18 14:47:49.394+08:00 wakuSwap init 2                            topics="wakunode" tid=10675
INF 2020-11-18 14:47:49.394+08:00 wakuSwap init 1                            topics="wakuswap" tid=10675
Hello
I am writing some stuff
So I can retrieve it later
Byebye
INF 2020-11-18 14:47:50.641+08:00 Hit store handler                          tid=10675
DBG 2020-11-18 14:47:50.641+08:00 Using SWAPAccounting query                 topics="wakunode" tid=10675
INF 2020-11-18 14:47:51.820+08:00 Accounting 2 for                           topics="wakuswap" tid=10675 peerId=16*3q4YjS n=4
INF 2020-11-18 14:47:51.820+08:00 Accounting test                            topics="wakuswap" tid=10675 text=test
[Chronicles] Log message not delivered: Accounting pre
key not found: 16Uiu2HAmJb2e28qLXxT5kZxVUUoJt72EMzNGXB47Rxx5hw3q4YjS
INF 2020-11-18 14:47:51.820+08:00 Accounting post                            topics="wakuswap" tid=10675 accounting=4
Hello

```

### CODE REVIEW

File: waku/v2/protocol/waku_swap.nim [hunk 1]
```diff
@@ -76,8 +76,33 @@ proc init*(T: type Cheque, buffer: seq[byte]): ProtoResult[T] =
...
+    info "NYI swap handle incoming connection"
+
+  proc accountFor(peerId: PeerId, n: int) {.gcsafe, closure.} =
+    info "Accounting for", peerId, n
+    info "Accounting test", text = wakuSwap.text
+    # Nicer way to write this?
+    if wakuSwap.accounting.hasKey(peerId):
+      wakuSwap.accounting[peerId] += n
+    else:
+      wakuSwap.accounting[peerId] = n
```
Comment (decanus) [inline waku/v2/protocol/waku_swap.nim:91]:
```suggestion
      wakuSwap.accounting[peerId] += n
```

@oskarth I feel like just this should work, but I may be wrong.

Comment (oskarth) [reply waku/v2/protocol/waku_swap.nim:91]: Nope, that is a compile error when I tried it IIRC

Comment (oskarth) [reply waku/v2/protocol/waku_swap.nim:91]: Or key not found error at runtime, forgot

Comment (decanus) [reply waku/v2/protocol/waku_swap.nim:91]: Ah sad. Then idk, other than that this looks fine to me

**Review by decanus — APPROVED** [review]
looks fine

**Review by jm-clius — APPROVED** [review]

**Pull request merged**